docs/testspec/src/12_pubkey_kem.rst

```diff
@@ -1,6 +1,124 @@
+   | **Description:**       | Uses the KAT vectors of Classic McElieces's reference implementation as |
+   |                        | specified in the NIST submission                                        |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Preconditions:**     | None                                                                    |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Input Values:**      | Test Vectors with RNG seed inputs in:                                   |
+   |                        |                                                                         |
+   |                        | * :srcref:`src/tests/data/pubkey/cmce_kat_hashed.vec`                   |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Expected Output:**   | Above described test vector files contain expected values for:          |
+   |                        |                                                                         |
+   |                        | * Classic McEliece Public Key                                           |
+   |                        | * Classic McEliece Private Key                                          |
+   |                        | * Ciphertext                                                            |
+   |                        | * Shared Secret                                                         |
+   +------------------------+-------------------------------------------------------------------------+
+   | **Steps:**             | For each KAT vector:                                                    |
+   |                        |                                                                         |
+   |                        | #. Seed a AES-256-CTR-DRBG with the specified RNG seed                  |
+   |                        |                                                                         |
+   |                        | #. Use the seeded RNG to generate a Classic McEliece key pair and       |
+   |                        |    compare it to the expected public and private key in the test        |
+   |                        |    vector. This uses the key encoding as implemented in the reference   |
+   |                        |    implementation and hashes the result using SHAKE-256(512) to save    |
+   |                        |    disk space in the KAT vectors.                                       |
+   |                        |                                                                         |
+   |                        | #. Check that the expected algorithm properties of the generated key    |
+   |                        |    match the generic expectations for KEMs (supports key encapsulation, |
```

Comment:
How do we check this? By just executing them once? Is this still only true for the SK? In that case, clarify

Reply:
The table **PKENC-CMCE-1** describes the generic KEM test case that is the same for all KEMs. I, therefore, more or less copy-pasted the test specification. I think it's ok to leave this part as it is for now. The properties are tested `For each KAT vector` as stated above. How exactly the specific properties are tested is too much detail, in my opinion. The only thing I'll change is writing in the plural `...of the generated keys` to include sk and pk.